### PR TITLE
[SPARK-44281][SQL] Move QueryCompilation error that used by DataType to sql/api as DataTypeErrors

### PR DESCRIPTION
**Pull request by amaliujia**
### What changes were proposed in this pull request?

Move QueryCompilation error that used by DataType to sql/api as DataTypeErrors.

### Why are the changes needed?

To further simplify DataType interface.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Exiting tests.

### CODE REVIEW

**Comment by amaliujia** [comment]
@cloud-fan

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryErrorsBase.scala [hunk 1]
```diff
@@ -92,10 +84,6 @@ private[sql] trait QueryErrorsBase {
     quoteByDefault(text.toUpperCase(Locale.ROOT))
   }
 
-  def toSQLConf(conf: String): String = {
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryErrorsBase.scala:95]: it's a bit weird that we put this in `DataTypeErrorsBase`, shall we name it `QuotingUtils`? And we can move methods to it, like `toSQLType`

Comment (amaliujia) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryErrorsBase.scala:95]: done

**Review by hvanhovell — APPROVED** [review]
LGTM

**Pull request closed without merging**